modules/analysis-common/src/test/java/org/opensearch/analysis/common/StemmerOverrideTokenFilterFactoryTests.java

```diff
@@ -88,7 +88,7 @@ public void testRulesOk() throws IOException {
     }
 
     public void testRulePartError() {
-        RuntimeException ex = expectThrows(RuntimeException.class, () -> create("a => 1", "b,c => 2", "# This is a comment", "=>a=>b"));
-        assertEquals("Line [4]: Invalid keyword override rule: =>a=>b", ex.getMessage());
```

Comment:
In 2.4, `"# This is a comment"` can be used as a mapping rule,  
But after this PR get merged, `"# This is a comment"` will not work any more.
Instead, we will need to use `"# => _hashtag_"`. 
I saw in `MappingCharFilterFactoryTests` and` StemmerOverrideTokenFilterFactoryTests`, `# This is a comment" `is removed and no longer tested. 

We should update tests files using the new mapping rule `"# => _hashtag_"` better than removing `# This is a comment"` to make sure the new mapping rule is working as expected.

Reply:
Updated

Reply:
Good call on this, looks to have caught an issue in `WordDelimiterGraphTokenFilterFactoryTests`. I'll have to take a look next week.

I personally think just merging the first commit would be safer here, which just fixes the regression in `char_filter` rather than removing comment support completely.

Reply:
ok